common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/indexedRecordOfStrings.ts

```diff
@@ -94,3 +94,40 @@ aImplicitType.length; // ok: aImplicitType is the continuation of the inferred t
 
 extendedIndexedRecordOfStrings.a.length; // ok: Accessing string property of extendedIndexedRecordOfStrings is allowed
 extendedIndexedRecordOfStrings.b.length; // defect: Accessing length of index property 'b', but 'b' might not be present
+
+interface NestedIndexSignatureType {
+	[FirstNestedKey: string]: {
+		[SecondNestedKey: string]: {
+			[ThirdNestedKey: string]: string;
+		};
+	};
+}
+
+function TestNullish(
+	base: NestedIndexSignatureType | undefined,
+	newData: NestedIndexSignatureType,
+): NestedIndexSignatureType {
+	const baseDatastore = base ?? {};
+	for (const [newDataKey, newDataValue] of Object.entries(newData)) {
+		const mergedData = baseDatastore[newDataKey] ?? {};
+		for (const valueManagerKey of Object.keys(newDataValue)) {
+			for (const [nestedDataKey, value] of Object.entries(newDataValue[valueManagerKey])) {
+				mergedData[valueManagerKey] ??= {};
+				const oldData = mergedData[valueManagerKey][nestedDataKey]; // ok: Accessing nested property nestedDataKey of mergedData[valueManagerKey] is allowed becauuse its assied using ??=
```

Comment:
```suggestion
				const oldData = mergedData[valueManagerKey][nestedDataKey]; // ok: Accessing nested property nestedDataKey of mergedData[valueManagerKey] is allowed because it's accessed using ??=
```